src/detect-engine-analyzer.c

```diff
@@ -861,6 +862,14 @@ static void DumpMatches(RuleAnalyzer *ctx, JsonBuilder *js, const SigMatchData *
                 jb_close(js);
                 break;
             }
+            case DETECT_ACK: {
+                const DetectAckData *cd = (const DetectAckData *)smd->ctx;
+
+                jb_open_object(js, "ack");
+                jb_set_uint(js, "ack", cd->ack);
```

Comment:
After considering our documentation, let's go with `number` for this inner field.

So the expectation is that for the output, we will see not only the keyword name, but also what was set in the rule for the ack number.

Reply:
OK.

I'll check it out—you mean, I should do `cd->number` right?

Reply:
No. There is no `cd->number`. Your current object would look like `ack.ack: 4`, you should instead make it `ack.number: 4` as per Juliana's comments. Does this make sense?